src/contract/ga/index.ts

```diff
@@ -0,0 +1,171 @@
+/*
+ * ISC License (ISC)
+ * Copyright (c) 2022 aeternity developers
+ *
+ *  Permission to use, copy, modify, and/or distribute this software for any
+ *  purpose with or without fee is hereby granted, provided that the above
+ *  copyright notice and this permission notice appear in all copies.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
+ *  REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+ *  AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
+ *  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+ *  LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
+ *  OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+ *  PERFORMANCE OF THIS SOFTWARE.
+ */
+
+/**
+ * Generalized Account module - routines to use generalized account
+ *
+ * @module @aeternity/aepp-sdk/es/contract/ga
+ * @export GeneralizedAccount
+ * @example import { GeneralizedAccount } from '@aeternity/aepp-sdk'
+ */
+
+// @ts-expect-error TODO: remove me
+import { TX_TYPE } from '../../tx/builder/schema'
+// @ts-expect-error TODO: remove me
+import { buildTx, unpackTx } from '../../tx/builder'
+import { prepareGaParams } from './helpers'
+import { hash } from '../../utils/crypto'
+import { decode, EncodedData } from '../../utils/encoder'
+// @ts-expect-error TODO remove
+import { getNetworkId } from '../../node'
+import { IllegalArgumentError, MissingParamError } from '../../utils/errors'
+
+export interface AuthData {
+  gasLimit: number
+  callData: EncodedData<'cb'>
+  source?: string
+  args?: object
+}
+
+interface GAObject {
+  owner: string
+  transaction: string
+  rawTx: string
+  gaContractId: string
+}
+
+/**
+ * @alias module:@aeternity/aepp-sdk/es/contract/ga
+ * Check if account is GA
+ * @param address - Account address
+ * @returns if account is GA
+ */
+export async function isGA ({ onAccount }: {onAccount: any}, address: string): Promise<boolean> {
+  const { contractId } = await onAccount.getAccount(address)
+  return contractId != null
+}
+
+/**
+ * Convert current account to GA
+ * @alias module:@aeternity/aepp-sdk/es/contract/ga
+ * @param authFnName - Authorization function name
+ * @param source - Auth contract source code
+ * @param args - init arguments
+ * @param options - Options
+ * @returns General Account Object
+ */
+export async function createGeneralizedAccount (
+  { onAccount }: { onAccount: any},
+  authFnName: string,
+  source: string,
+  args: any[] = [],
+  options: object = {}):
```

Comment:
Its type should be a combination of options types of functions where it gets passed, like
```suggestion
  options: Parameters<Account['send']>[1] & Parameters<gaAttachTx>[0] = {}):
```